lib/chef/resource.rb

```diff
@@ -943,8 +954,63 @@ def provider_base(arg=nil)
         end
         @provider_base ||= arg || Chef::Provider
       end
-    end
 
+      #
```

Comment:
Would you mind changing this from `class << self` to `self.method` style? Took me too much work to figure out why there are two allowed actions methods. There is also some weird stuff with ivar setting and method lookup that happens in `class << self`, which isn't a problem yet, but recently bit me on a different project.